tests/providers/mysql/hooks/test_mysql.py

```diff
@@ -184,6 +197,20 @@ def setUp(self):
         self.db_hook.get_connection = mock.Mock()
         self.db_hook.get_connection.return_value = self.connection
 
+        class FakeMySQLConnection(object):
```

Comment:
since airflow is python 3+ you can remove `(object)`

Reply:
I've reworked this to have the fake object extend the `MagicMock` class so that the property checks in `mysql.py` pass and so that the mock assertions can be used.